src/DependencyInjection/DoctrineExtension.php

```diff
@@ -300,8 +300,15 @@ protected function loadDbalConnection($name, array $connection, ContainerBuilder
             $def->setClass($options['wrapperClass']);
         }
 
-        if (! empty($connection['use_savepoints'])) {
-            $def->addMethodCall('setNestTransactionsWithSavepoints', [$connection['use_savepoints']]);
+        if (isset($connection['use_savepoints'])) {
+            // DBAL >= 4 always has savepoints enabled. So we only need to call "setNestTransactionsWithSavepoints" for DBAL < 4
+            if (method_exists(Connection::class, 'getEventManager')) {
+                if ($connection['use_savepoints']) {
```

Comment:
due to `empty` it was actually never calling `setNestTransactionsWithSavepoints(false)`. So I kept this behavior the same